client/trino-jdbc/pom.xml

```diff
@@ -64,6 +64,16 @@
             <artifactId>okhttp</artifactId>
         </dependency>
 
+        <dependency>
+            <groupId>io.opentelemetry</groupId>
+            <artifactId>opentelemetry-api</artifactId>
+        </dependency>
+
+        <dependency>
+            <groupId>io.opentelemetry.instrumentation</groupId>
```

Comment:
We might not want to add this as a dependency, and we can't shade it. I'll see if we can use reflection to check if this is available, and maybe the OTEL Java agent supports some mapping of class namespaces.

Reply:
Agreed, we can check for `OpenTelemetry` or any class in the API, and also check for `Context` just to be safe. Then do the integration manually.

Reply:
I couldn't make it work with reflection. The instrumentation couldn't be applied to the shaded okhttp client. Instead, I shaded the instrumentation too, assume the OTEL API is provided, and skip instrumenting if it's not. Do you think that can work? Is there any reason the OTEL API used this way could cause conflicts at runtime?

Reply:
![image](https://user-images.githubusercontent.com/795177/234947608-8a526e4d-a2b7-4e74-bd85-0f482e9800af.png)

Reply:
I know we can do the instrumentation manually, but I don't understand why would we have to use reflection for OTEL API.